plugin/src/main/java/org/opensearch/ml/settings/MLCommonsSettings.java

```diff
@@ -77,6 +77,16 @@ private MLCommonsSettings() {}
     public static final Setting<Integer> ML_COMMONS_JVM_HEAP_MEM_THRESHOLD = Setting
         .intSetting("plugins.ml_commons.jvm_heap_memory_threshold", 85, 0, 100, Setting.Property.NodeScope, Setting.Property.Dynamic);
 
+    public static final Setting<Integer> ML_COMMONS_DISK_FREE_SPACE_THRESHOLD = Setting
+        .intSetting(
+            "plugins.ml_commons.disk_free_space_threshold",
+            5,
+            0,
+            Integer.MAX_VALUE,
+            Setting.Property.NodeScope,
+            Setting.Property.Dynamic
+        );
```

Comment:
nit: another alternative could be `ByteSizeValue` instead of int. it has more fine-grained control.